5task Avvakumov.cpp

```diff
@@ -0,0 +1,155 @@
+#include <iostream>
+#include <map>
+#include <set>
+#include <sstream>
+#include <iomanip>
+#include <stdexcept>
+
+class Date {
+public:
+    Date(const std::string& dateStr) {
+        std::istringstream ss(dateStr);
+        char delimiter;
+        ss >> m_year >> delimiter >> m_month >> delimiter >> m_day;
+        validateDate(ss, delimiter, dateStr);
+    }
+
+    int GetYear() const {
+        return m_year;
+    }
+
+    int GetMonth() const {
+        return m_month;
+    }
+
+    int GetDay() const {
+        return m_day;
+    }
+
+    friend bool operator<(const Date& lhs, const Date& rhs) {
```

Comment:
Можно сделать без friend принимая только объект, с которым будем сравнивать
(доступ к его полям в этом случае будет, несмотря на то, что они private)
```suggestion
    bool operator<(const Date& other) {
```
Вообще friend - зло, применять его нужно когда без него вообще никак